metrics/wait.io.table/table.go

```diff
@@ -223,11 +267,57 @@ func (t *Table) Collect(ctx context.Context, levelName string) ([]blip.MetricVal
 	}
 
 	if o.truncate {
-		_, err := t.db.Exec(TRUNCATE_QUERY)
+		trCtx, cancelFn := context.WithTimeout(ctx, o.truncateTimeout)
+		defer cancelFn()
+		_, err := t.db.ExecContext(trCtx, TRUNCATE_QUERY)
 		if err != nil {
-			return nil, err
+			return t.collectTruncateError(err, o, metrics)
+		} else if o.hadTruncateError {
+			// Upon success we can start emitting metrics normally again but we need to
+			// wait for the next collect for the truncate to have taken effect.
+			// Emit whatever metrics would be emitted based on the error policy.
+			metrics, err = t.collectTruncateError(fmt.Errorf("Truncate successful, waiting for next metric collection."), o, metrics)
 		}
+
+		o.hadTruncateError = false
 	}
 
 	return metrics, err
 }
+
+func (t *Table) collectTruncateError(err error, options *tableOptions, collectedMetrics []blip.MetricValue) ([]blip.MetricValue, error) {
```

Comment:
Is it possible this func and the one for the other collector can be factored out into common code somewhere? No big deal for now; just an idea in case future collectors `TRUNCATE` and need this same error policy logic.

Reply:
Yes. They are pretty much the same and I can make a common version. Is there a place that makes sense to put that?